[Process] Added the RFC template docs [#2594] (#2601)
delta-io/delta: .github/ISSUE_TEMPLATE/protocol-rfc.md

---
name: Protocol Change Request

> about: Use this template for reporting a bug

Reviewer: ```suggestion
about: Use this template to propose a new feature that impacts the Delta protocol specification
```

labels: 'protocol'
title: '[PROTOCOL RFC]'

---

## Protocol Change Request

### Description of the protocol change

<!--
Please describe the motivation and high-level description of the protocol change you are proposing.
For a fairly large protocol change, it is recommended that you provide a design doc - (e.g., a google doc, preferably with the ability to comment in the doc). 

> For the next steps on how to proceed with the request, see the protocol RFC process in https://github.com/delta-io/delta/tree/master/protocol_rfcs

Reviewer: ```suggestion
For the next steps on how to proceed with the request, see the protocol RFC process in [https://github.com/delta-io/delta/tree/master/protocol_rfcs](https://github.com/delta-io/delta/tree/master/protocol_rfcs)
```
^^ should turn it into a convenient hyperlink

Reviewer: Actually... it will probably render as a hyperlink automatically, just not in the PR diff 🤦

Author: Actually neither. Its inside a comment, so it wont render in the markdown form at all. :)

--> 

### Willingness to contribute

> The Delta Lake Community encourages bug fix contributions. Would you or another member of your organization be willing to contribute a fix for this bug to the Delta Lake code base?

Reviewer: ```suggestion
The Delta Lake Community encourages protocol innovations. Would you or another member of your organization be willing to contribute this feature to the Delta Lake code base?
```
